test/validation/api/ipsec/ipsec.c

```diff
@@ -866,6 +878,38 @@ static int ipsec_send_out_one(const ipsec_test_part *part,
 	return num_out;
 }
 
+int ipsec_test_sa_update_seq_num(odp_ipsec_sa_t sa, uint32_t seq_num)
+{
+	odp_ipsec_test_sa_operation_t sa_op;
+	odp_ipsec_test_sa_param_t sa_param;
+
+	sa_op = ODP_IPSEC_TEST_SA_UPDATE_SEQ_NUM;
+	sa_param.seq_num = seq_num;
+
+	return odp_ipsec_test_sa_update(sa, sa_op, sa_param);
+}
+
+static void ipsec_pkt_seq_num_check(odp_packet_t pkt, uint32_t seq_num)
+{
+	uint32_t l3_off = odp_packet_l3_offset(pkt);
+	uint32_t l4_off = odp_packet_l4_offset(pkt);
```

Comment:
This is incorrect. The value of l4 offset is undefined after outbound IPsec operation.

It would also be good to check whether the odp_packet_l3_offset() call succeeds.

Reply:
Agreed. Updated the function to compute l4_off as l3_off + IPv4 Header length.

Also added check for  return value of odp_packet_l3_offset().